src/main/java/org/springframework/test/context/junit5/SpringParameterExtension.java

```diff
@@ -0,0 +1,65 @@
+package org.springframework.test.context.junit5;
+
+import org.junit.gen5.api.extension.ExtensionContext;
+import org.junit.gen5.api.extension.MethodInvocationContext;
+import org.junit.gen5.api.extension.MethodParameterResolver;
+import org.junit.gen5.api.extension.ParameterResolutionException;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.context.TestContext;
+import org.springframework.test.context.TestContextManager;
+import org.springframework.util.ReflectionUtils;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+
+/**
+ * @author Tadaya Tsuyukubo
+ */
+public class SpringParameterExtension extends SpringExtension implements MethodParameterResolver {
+
+	@Override
+	public boolean supports(Parameter parameter, MethodInvocationContext methodInvocationContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+		return true;  // TODO: should check
+	}
+
+	@Override
+	public Object resolve(Parameter parameter, MethodInvocationContext methodInvocationContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+
+		ApplicationContext applicationContext = getApplicationContext(extensionContext.getTestClass());
+
+		// special case for application context
+		if (ApplicationContext.class.isAssignableFrom(parameter.getType())) {
+			return applicationContext;
+		}
+
+		// TODO: need to resolve beans more sophisticated way. for example, generics support
+
+		Object bean;
+
+		Qualifier qualifier = parameter.getAnnotation(Qualifier.class);
+		if (qualifier != null) {
+			bean = applicationContext.getBean(qualifier.value());
+			if (bean != null) {
+				return bean;
+			}
+		}
+
+		bean = applicationContext.getBean(parameter.getType());
+		if (bean != null) {
+			return bean;
+		}
+		return applicationContext.getBean(parameter.getName());
+	}
```

Comment:
Unfortunately, this technique is inappropriate for injecting specific beans from the `ApplicationContext`.

The reason is that we cannot _compete_ with other registered method parameter resolvers.

The only viable solution is a Spring-specific annotation (other than `@Qualifier` on its own) for performing dependency injection for an individual method parameter. Unfortunately, `@Autowired` cannot _yet_ be used like that.

But... fear not: I have been aware of this for several months, and we should be getting first-class support for this in core Spring in the 4.3 or 5.0 timeline.

For the time being, we simply have to live without dependency injection for individual beans into method parameters. Field injection via `@Autowired` or injection of the `ApplicationContext` into a test method will therefore have to suffice for now.